app/models/base.py

```diff
@@ -9,3 +13,44 @@ class SoftDeletionModel(db.Model):
     __abstract__ = True
 
     deleted_at = db.Column(db.DateTime(timezone=True))
+
+    def get_new_identifier(self):
+        """
+        Factory function that calls the model-appropriate method and returns an identifier
+        :return:
+        """
+        model_name = type(self).__name__
+        with db.session.no_autoflush:
+            if model_name in {'Order', 'EventInvoice'}:
+                identifier = self.random_uuid()
+            elif model_name in {'SocialLink', 'Event'}:
+                identifier = self.random_string_identifier()
+            else:
+                identifier = None
```

Comment:
This is not how to do things, there is no need of if else, just use the correct method in correct model